service/geopmdpy/system_files.py

```diff
@@ -161,25 +162,28 @@ def secure_make_file(path, contents):
 
     The file is created by first writing the output to a temporary
     file with restricted permissions (mode 0o600) within the same
-    directory, and then renaming the file after it is closed.  This
-    temporary file with have the path `<path>-<uuid>-tmp`.
+    directory, and then renaming the file after it is closed.
 
     The rename operation is atomic, and will overwrite any existing
     file located in the specified path.  The permissions on the output
     file are mode 0o600 with uid and gid matching the process values.
 
+    The temporary file has the prefix `tmp.`. This file may persist if
+    the program exits abruptly.
+
     Args:
         path (str): The path where the file is created
 
         contents (str): The contents of the created file
 
     """
-    temp_path = f'{path}-{uuid.uuid4()}-tmp'
+    directory = os.path.dirname(os.path.abspath(path))
+    prefix = f'{os.path.basename(path)}-tmp-'
     old_mask = os.umask(0o077)
```

Comment:
This existing call to umask had no impact on the old code, it also has no impact on the new code.  In both cases the open() call is made with mode set to 0o600, so masking off the "other" and "group" modes has no impact.  If we wanted to remove the umask() call we could, but it also does no harm.